source/Calamari/Kubernetes/Conventions/ResourceStatusReportConvention.cs

```diff
@@ -18,7 +19,11 @@ public ResourceStatusReportConvention(ResourceStatusReportExecutor statusReportE
 
         public void Install(RunningDeployment deployment)
         {
-            statusReportExecutor.ReportStatus(deployment.CurrentDirectory, commandLineRunner, deployment.EnvironmentVariables);
+            var successful = statusReportExecutor.ReportStatus(deployment.CurrentDirectory, commandLineRunner, deployment.EnvironmentVariables);
+            if (!successful)
+            {
+                throw new TimeoutException("Not all resources have deployed successfully within timeout");
+            }
```

Comment:
I think it is better to communicate the result of the status check execution via the return value rather than an exception. But I noticed that the exception is necessary here for it to work via conventions, so I moved the previous exception here. @scottmerchant please call out if there are something I missed here, thanks

Reply:
Just FYI some of this is going to conflict with my changes. Can we hold off on merging this? I'll help resolve the conflicts after I've merged my things but it might take a few days.

Reply:
Main reason is I'm going on leave soon and I really want to get my stuff in before I go... 🙏

Reply:
Thanks @scottmerchant for the info. No problems, but I wonder if the conflict is from changing from exception to return value? If so, will it prevent the conflict if I change this back to the existing mechanism, just add more logs?

Reply:
Yeah if you kept it the same with just some extra logs it would be ok for now. We can do the change to use return values later if you want. In the new future I'm making changes to remove the `ResourceStatusReportConvention` because it won't be executed via an Install convention then...

Reply:
But I would still hold off merging for the moment please.

Reply:
Thanks, I will hold off merging. In the meantime, since refactoring the return value isn't the goal for this PR, I will just revert that part and keep the changes only to logging.

Reply:
I've reverted that part of refactoring. I will add you as a reviewer to this PR and let me know when you think this PR is good to be merged @scottmerchant